src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -5134,6 +5134,16 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     inline void visit_expr_wrapper(ASR::expr_t* x, bool load_ref=false) {
+        // Check if *x is nullptr.
+        if( x == nullptr ) {
+            throw CodeGenError("Internal error: x is nullptr");
+        }
+
+        // Check if the type of *x is expr_t or not.
+        if( !ASR::is_a<ASR::expr_t>(*x) ) {
+            throw CodeGenError("Internal error: x is not of type expr_t");
+        }
```

Comment:
I think `x` has to be an `expr_t`, since that's the type of the argument. Also the `is_a` check works for the end nodes like IntegerBinOp; it works for expr if `x` was of type `asr_t`. 

I would remove this last check.

Reply:
This has been resolved.